iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/AnonymousAccessToResourceCheck.java

```diff
@@ -64,7 +64,8 @@ protected void registerResourceConsumer() {
   }
 
   private static void checkWebSites(ContextualResource resource) {
-    ContextualResource authSettingsV2 = resource.childResourceBy("config", it -> isValue(it.name(), WEBSITES_CONFIG_AUTH_SETTINGS_V2_RESOURCE_NAME).isTrue());
+    ContextualResource authSettingsV2 = resource.childResourceBy("config", it -> isValue(it.name(), WEBSITES_CONFIG_AUTH_SETTINGS_V2_RESOURCE_NAME).isTrue() ||
+      isValue(it.name(), resource.name + "/" + WEBSITES_CONFIG_AUTH_SETTINGS_V2_RESOURCE_NAME).isTrue());
```

Comment:
I didn't find a good way to hide this details from the predicate. With `type`, it's easier, because type is passed as a string and we can transform it during processing. However, for the same predicate to be applicable to an external child resource, we would need to create a synthetic copy of the node with trimmed name prefix.

Reply:
Could you extract those conditions to separate method, as it is a long condition?